client/reader/stream/refresh-post.jsx

```diff
@@ -102,6 +102,9 @@ export default class ReaderPostCardAdapterFluxContainer extends React.Component
 				blogId: post.discover_metadata.featured_post_wpcom_data.blog_id,
 				postId: post.discover_metadata.featured_post_wpcom_data.post_id
 			} );
+			if ( originalPost ) {
+				originalPost.referral = { blogId: DiscoverHelper.discoverBlogId, postId: post.ID }
```

Comment:
That function _does_ return `{ originalPost }` but I think I see the issue. `referral` doesn't need to be part of the state but it does need to get attached to the `originalPost` at some point. I did that in the older `./post.jsx` in the click handler but did not do that here ( I fixed the state mutation there as you correctly suggested ). 

`DRY` failure on my end, so to speak. In post.jsx we do the `isDiscover` logic in click handler and not in the state related functions. In the refresh the opposite is true so I incorrectly _DRY_'d the `isDiscover` logic and set the referral in the state management flow. 

Thanks for catching this!